File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingScheduleImportColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingScheduleImportRow } from "./TrainingScheduleImportRow";

export interface TrainingScheduleImportColumns {
    TrainingScheduleImportId: Column<TrainingScheduleImportRow>;
    TrainingTerm: Column<TrainingScheduleImportRow>;
    TrainingUnit: Column<TrainingScheduleImportRow>;
    Department: Column<TrainingScheduleImportRow>;
    TrainingType: Column<TrainingScheduleImportRow>;
    TrainerName: Column<TrainingScheduleImportRow>;
    TrainerNumber: Column<TrainingScheduleImportRow>;
    Day: Column<TrainingScheduleImportRow>;
    Time: Column<TrainingScheduleImportRow>;
    LectureCount: Column<TrainingScheduleImportRow>;
    Course: Column<TrainingScheduleImportRow>;
    IsActive: Column<TrainingScheduleImportRow>;
}

export class TrainingScheduleImportColumns extends ColumnsBase<TrainingScheduleImportRow> {
    static readonly columnsKey = 'Administration.TrainingScheduleImport';
    static readonly Fields = fieldsProxy<TrainingScheduleImportColumns>();
}

File: QadeerApp.Web/Modules/Administration/TrainingScheduleImport/TrainingScheduleImportColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingScheduleImport")]
[BasedOnRow(typeof(TrainingScheduleImportRow), CheckNames = true)]
public class TrainingScheduleImportColumns
{
    [EditLink, AlignRight]
    public int TrainingScheduleImportId { get; set; }

    [EditLink, Width(160)]
    public string TrainingTerm { get; set; }

    [Width(140)]
    public string TrainingUnit { get; set; }

    [Width(140)]
    public string Department { get; set; }

    [Width(120)]
    public string TrainingType { get; set; }

    [Width(120)]
    public string TrainerName { get; set; }

    [Width(100)]
    public int TrainerNumber { get; set; }

    [Width(90)]
    public string Day { get; set; }

    [Width(120)]
    public string Time { get; set; }

    [Width(90)]
    public int LectureCount { get; set; }

    [Width(140)]
    public string Course { get; set; }

    [Width(90)]
    public short IsActive { get; set; }
}
